### PR TITLE
chore: feat(eg|dms|utils): add new resource and update the some parameter and function logic

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
1. Support a new event stream resource for EG service.
2. Change the `ssl_enable` parameter behavior from attribute to argument.
3. Update the verification logic of some kafka resource parameters.
4. Update the function logic for public ignore function.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. ignore the empty values of the type map.
2. change the verification logic of some parameters.
3. add new event stream resource support.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run=TestAccEventStream_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run=TestAccEventStream_basic -timeout 360m -parallel 4
=== RUN   TestAccEventStream_basic
=== PAUSE TestAccEventStream_basic
=== CONT  TestAccEventStream_basic
--- PASS: TestAccEventStream_basic (862.11s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        862.152s
```
```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run=TestAccEventStream_rocketMQ'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run=TestAccEventStream_rocketMQ -timeout 360m -parallel 4
=== RUN   TestAccEventStream_rocketMQ
=== PAUSE TestAccEventStream_rocketMQ
=== CONT  TestAccEventStream_rocketMQ
--- PASS: TestAccEventStream_rocketMQ (769.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        769.945s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run=TestAccEventStream_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run=TestAccEventStream_basic -timeout 360m -parallel 4
=== RUN   TestAccEventStream_basic
=== PAUSE TestAccEventStream_basic
=== CONT  TestAccEventStream_basic
--- PASS: TestAccEventStream_basic (844.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        844.707s
```

File: huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_event_stream_test.go [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+					resource.TestCheckResourceAttr(rName, "action", "PAUSE"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"source",
+					"sink",
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_event_stream_test.go:397]: if there parameters are not returned by api, should add a description about it in doc

File: huaweicloud/utils/type_convert.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func ValueIngoreEmpty(v interface{}) interface{} {
 		return nil
 	}
 
-	if (vl.Kind() == reflect.Array || vl.Kind() == reflect.Slice) && vl.Len() == 0 {
+	if (vl.Kind() == reflect.Array || vl.Kind() == reflect.Slice || vl.Kind() == reflect.Map) && vl.Len() == 0 {
```
Comment (Jason-Zhang9309) [inline huaweicloud/utils/type_convert.go]: looks empty map will be ignored in utils.RemoveNil,
so this change is not necessary

File: huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_event_stream_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_event_stream_test.go:397]: yes and finished.

File: huaweicloud/utils/type_convert.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/utils/type_convert.go]: sure, and this commit has been removed.

**Pull request merged**